core/src/main/java/com/linecorp/armeria/client/endpoint/healthcheck/PartialHealthCheckStrategy.java

```diff
@@ -0,0 +1,185 @@
+/*
+ * Copyright 2019 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.client.endpoint.healthcheck;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Random;
+import java.util.Set;
+import java.util.concurrent.ThreadLocalRandom;
+
+import org.jctools.maps.NonBlockingHashSet;
+
+import com.linecorp.armeria.client.Endpoint;
+
+import com.google.common.collect.ImmutableSet;
+
+/**
+ * A strategy to check part of candidates by health.
+ */
+public class PartialHealthCheckStrategy implements HealthCheckStrategy {
+
+    private final Set<Endpoint> selectedCandidates;
+    private final Set<Endpoint> unhealthyCandidates;
+    private final TargetCount max;
+    private Set<Endpoint> candidates;
+
+    /**
+     * Creates a new instance.
+     */
+    PartialHealthCheckStrategy(TargetCount max) {
+        this.max = requireNonNull(max, "max");
+        selectedCandidates = new NonBlockingHashSet<>();
+        unhealthyCandidates = new NonBlockingHashSet<>();
+        candidates = ImmutableSet.copyOf(new ArrayList<>());
+    }
+
+    private static Set<Endpoint> selectRandomly(Set<Endpoint> availableCandidates, int cnt) {
```

Comment:
Could be moved after `updateSelectedCandidates()` for readability.

Reply:
fixed